ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -371,6 +380,38 @@ IndexValue find(StoreKey keyToFind) throws StoreException {
     return toReturn;
```

Comment:
Since the returned set is for read-only purposes, you could wrap this in a `Collections.unmodifiableNavigableSet()`

Reply:
Yeah - that's a good idea. Will make the change

Reply:
Done